packages/parser/bin/index.ts

```diff
@@ -0,0 +1,32 @@
+import 'reflect-metadata';
+
+import { Amqp } from '@spectacles/brokers';
+import { AmqpResponseOptions } from '@spectacles/brokers/typings/src/Amqp';
+import { on } from 'events';
+import { Message } from '@spectacles/types';
+import * as postgres from 'postgres';
+import { Lexer, Parser, Tokens, Unordered } from 'lexure';
+
+const broker = new Amqp('gateway');
+const pg = postgres();
+
+(async () => {
+	await broker.connect('rabbitmq');
+	await broker.createQueue('COMMAND');
+	await broker.subscribe(['MESSAGE_CREATE']);
+	
+	for await (const [message, { ack }] of on(broker, 'MESSAGE_CREATE') as AsyncIterableIterator<[Message, AmqpResponseOptions]>) {
+		ack();
+		const [{ prefix }] = await pg`select settings ->> 'PREFIX' as prefix
+			from settings
+			where guild_id = ${message.guild_id}
+				is not null;` as [{ prefix: string | null }];
+		const lexer = new Lexer(message.content).setQuotes([['"', '"'], ['“', '”']]);
+		const tokens = lexer.lex();
+		const parser = new Parser(tokens).setUnorderedStrategy(Unordered.longShortStrategy());
+		const res = parser.parse();
+		const command = Tokens.extractCommand(s => s.startsWith(prefix ?? '?') ? 1 : null, res.ordered);
```

Comment:
The callback should return the length of the prefix, not always 1. Otherwise the command name won't be extracted correctly.

Reply:
That's what happens if you expect your usual prefix to only ever be 1 character 😅